receiver/m365receiver/logs.go

```diff
@@ -136,7 +136,9 @@ func (l *m365LogsReceiver) Start(ctx context.Context, host component.Host) error
 
 func (l *m365LogsReceiver) Shutdown(ctx context.Context) error {
 	l.logger.Debug("shutting down logs receiver")
-	l.cancel()
+	if l.cancel != nil {
+		l.cancel()
+	}
 	l.wg.Wait()
 	return l.checkpoint(ctx)
```

Comment:
In `checkpoint`, could we add a check for if `l.record` is nil? It wasn't obvious to me how we avoid clobbering the old checkpoint with the nil record in this scenario.